src/Php/ClassGenerator.php

```diff
@@ -178,6 +179,12 @@ private function handleSetter(Generator\ClassGenerator $generator, PHPProperty $
             $parameter->setDefaultValue(null);
         }
 
+        if (($parameter->getDefaultValue() instanceof ValueGenerator) &&
+            $parameter->getDefaultValue()->getValue() === null && $parameter->getType() !== null &&
+            substr($parameter->getType(), 0, 1) !== '?') {
+            $parameter->setType('?' . $parameter->getType());
```

Comment:
To generate
```
public function setSomething(?array $something= null)
```
instead of
```
public function setSomething(array $something= null)
```